qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
Note that we cannot arbitrarily change the sample ids when we want, as the
demultiplexed files and the biom tables will include those sample ids...

Let's see what other Qiita devs think about this

2015-02-11 14:17 GMT-08:00 Daniel McDonald <notifications@github.com>:

> In qiita_db/test/test_metadata_template.py
> <https://github.com/biocore/qiita/pull/843#discussion_r24539391>:
>
> > @@ -652,6 +775,133 @@ def test_create(self):
> >                 ['2.Sample3', "Value for sample 3"]]
> >          self.assertEqual(obs, exp)
> >
> > +    def test_create_int_prefix(self):
> > +        """Creates a new SampleTemplate"""
> > +        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
> > +        # The returned object has the correct id
> > +        self.assertEqual(st.id, 2)
> > +
> > +        # The relevant rows to required_sample_info have been added.
> > +        obs = self.conn_handler.execute_fetchall(
> > +            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
> > +        # sample_id study_id physical_location has_physical_specimen
> > +        # has_extracted_data sample_type required_sample_info_status_id
> > +        # collection_timestamp host_subject_id description
> > +        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
>
> Right, so you need external state maintained by a separate process in
> order to know how to interpret the variable -- resident memory, the
> variable, and its type are insufficient. That sucks as it opens the door
> for bugs and race conditions.
>
> If the downstream issue is producing valid qiime mapping files, then we
> can at least delay the construction of a valid qiime sample ID until the
> last moment, and represent these data internally in a manner that reflects
> their aggregate nature.
>
> If this ambiguity doesn't bother anyone else, then lets move on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/843/files#r24539391>.
>



-- 
Jose Navas